### ISSUE TITLE
Resolve `k3d` bug in the `kubefirst-api` script

### ISSUE THREAD

**Issue by alechp** [ssotops/k1space#62]
Currently, the output looks like this:
<img width="2519" alt="Screenshot 2024-08-29 at 11 28 54 AM" src="https://github.com/user-attachments/assets/f592bb82-7d56-4bba-9780-ad6dd3189e3f">

Issue is that it's attempting to run k3d cluster and it's not working.

Relevant function:
```go
func runKubefirstAPI(repoDir, logsDir string) {
	apiDir := filepath.Join(repoDir, "kubefirst-api")
	logFile := filepath.Join(logsDir, "kubefirst-api.log")

	// Check if k3d is installed
	if _, err := exec.LookPath("k3d"); err != nil {
		log.Error("k3d is not installed or not in PATH. Please install k3d and try again.")
		return
	}

	setupScript := `
#!/bin/bash
set -e
go install github.com/air-verse/air@latest
go install github.com/swaggo/swag/cmd/swag@latest
k3d cluster create dev || echo "Cluster 'dev' may already exist, continuing..."
k3d kubeconfig write dev
export K1_LOCAL_DEBUG=true
export K1_LOCAL_KUBECONFIG_PATH=$(k3d kubeconfig get dev)
export CLUSTER_ID="local-dev"
export CLUSTER_TYPE="k3d"
export INSTALL_METHOD="local"
export K1_ACCESS_TOKEN="local-dev-token"
export IS_CLUSTER_ZERO=true
if [ ! -f .env ]; then
    cp .env.example .env
fi
source .env
kubectl create namespace kubefirst --dry-run=client -o yaml | kubectl apply -f -
kubectl create secret generic kubefirst-clusters --from-literal=clusters='{}' -n kubefirst --dry-run=client -o yaml | kubectl apply -f -
kubectl create secret generic kubefirst-catalog --from-literal=catalog='{}' -n kubefirst --dry-run=client -o yaml | kubectl apply -f -
make updateswagger
make build
`

	setupCmd := exec.Command("bash", "-c", setupScript)
	setupCmd.Dir = apiDir

	err := runAndLogCommand(setupCmd, logFile, color.FgGreen)
	if err != nil {
		log.Error("Error setting up kubefirst-api", "error", err)
		return
	}

	airCmd := exec.Command("air")
	airCmd.Dir = apiDir

	err = runAndLogCommand(airCmd, logFile, color.FgGreen)
	if err != nil {
		log.Error("Error running kubefirst-api", "error", err)
	}
}
```

You can see we're inlining the script that (works locally & is mentioned in #30)

**Comment by alechp** [comment]
Thought it was working for a moment but realized that k3d was already running:

<img width="1148" alt="Screenshot 2024-08-29 at 4 13 54 PM" src="https://github.com/user-attachments/assets/452d48cb-5b11-4854-9297-f8174906960a">
<img width="838" alt="Screenshot 2024-08-29 at 4 14 03 PM" src="https://github.com/user-attachments/assets/9b2f9549-c98a-4c6e-bd47-5592c081b5a0">

**Comment by alechp** [comment]
resolved as of https://github.com/ssotops/k1space/pull/67

Also fixes the unfinished execution portion of #30
